Add python.imread & python.mahotas

luispedro opened the pull request:
These are two packages I developed and now myself and  many other regularly use.

luispedro committed:
python.imread: init at 0.5.1

This is a package to read image files into numpy arrays.

luispedro committed:
python.mahotas: init at 1.4.0

This is a computer vision package for Python based on numpy

FRidh commented:
Hi @luispedro, thanks for your contribution. Nice to see other scientists here adding Python packages.
Could you update the commit messages according to the nixpkgs guide.
http://nixos.org/nixpkgs/manual/#chap-submitting-changes

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -3097,6 +3097,35 @@ in modules // {
+            scipy
+          ];
+    propagatedBuildInputs = with self; [
+            python
```
no need to add python here

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -4940,6 +4969,36 @@ in modules // {
+            nose
+          ];
+    propagatedBuildInputs = with self; [
+            python
```
nor here

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -3097,6 +3097,35 @@ in modules // {
+      homepage = https://readthedocs.org/projects/mahotas/;
+      maintainers = with maintainers; [ luispedro ];
+      license = licenses.mit;
+      platforms = platforms.linux;
```
Is darwin/osx intentionally not supported?

luispedro commented:
Thanks. This is my first attempt at packaging things and I'm still getting the hang of it. I updated the PR according to your comments.

Cheers, Luis

FRidh commented:
cc @domenkozar

luispedro committed:
chessx: init at 1.2.2

Chessx is a chess database GUI. If an engine such as stockfish is
installed, it can also provide access to analysis.

domenkozar commented on pkgs/top-level/python-packages.nix:
```diff
@@ -3097,6 +3097,34 @@ in modules // {
+    propagatedBuildInputs = with self; [
+            numpy
+            imread
+          ];
```
(nitpick) please indent with two spaces

domenkozar commented on pkgs/top-level/python-packages.nix:
```diff
@@ -3097,6 +3097,34 @@ in modules // {
+
+    buildInputs = with self; [
+            nose
+            pillow
```
Is `pillow` not a runtime dependency?

luispedro replied:
No, just used in the test suite.

domenkozar commented on pkgs/top-level/python-packages.nix:
```diff
@@ -4940,6 +4968,35 @@ in modules // {
+            pkgs.libjpeg
+            pkgs.libpng
+            pkgs.libtiff
+            pkgs.libwebp
```
these packages shouldn't be propagated, they're just `buildInputs`

luispedro committed:
stockfish: init at 6

luispedro commented:
Accidentally closed and github won't let me reopen.
